### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by aariful-islam**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,6 +18,7 @@ function App() {
       <Route path="/confirm-password" element={<ConfirmPassword />} />
       <Route path="/project/invite" element={<InviteMember />} />
       <Route path="/projects" element={<ProjectList />} />
+      <Route path="/login" element={<LogIn></LogIn>} />
```
Comment (Nayeem-XTREME) [inline src/App.js]: Wrong style

File: src/state/LogIn/loginSlice.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+import { privatePost } from "../../utilities";
+export const fetchUserData = createAsyncThunk(
+  "user/signin",
+  async (user, { rejectWithValue }) => {
+    try {
+      const response = await privatePost("/signin", user);
```
Comment (Nayeem-XTREME) [inline src/state/LogIn/loginSlice.js]: Login is not private. User will get the token by login. User won't have it before login.

File: src/state/LogIn/loginSlice.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      if (response) {
+        return response.data;
+      }
```
Comment (Nayeem-XTREME) [inline src/state/LogIn/loginSlice.js]: No `if` required. All api request has either `response` or `error`.

**Review by Nayeem-XTREME — CHANGES_REQUESTED** [review]

File: src/pages/LogIn/index.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  };
+  return (
+    <div className="h-screen bg-cyan-50">
+      <div className="leftImage">
+        <img src={leftImage} alt="background" />
+      </div>
+
+      <div className="logo-bg">
+        <img
+          className={`logo h-32 w-48 bg-blue-800 object-contain p-3`}
```
Comment (Nayeem-XTREME) [inline src/pages/LogIn/index.js]: No need for ``{}``. Use ""

File: src/pages/LogIn/index.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+              value={password}
+              name="password"
+              placeholder="Enter Password"
+              onChange={(e) => setPassword(e.target.value)}
+            />
+
+            <div className="mt-6 ">
+              <button
+                onClick={handleFormSubmit}
+                style={{ width: "95%" }}
```
Comment (Nayeem-XTREME) [inline src/pages/LogIn/index.js]: Why inline?

File: src/pages/LogIn/index.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                className={` border-green focus:border-green-dark w-96 rounded border bg-blue-800 py-2 px-4 text-sm text-white focus:outline-none`}
+              >
+                Login
+              </button>
+            </div>
+            <h1 className={`py-4 text-center text-green-500`}>OR</h1>
+            <hr />
+            <div>
+              <p className={`text-center text-blue-700`}>
+                Can't log in?{" "}
```
Comment (Nayeem-XTREME) [inline src/pages/LogIn/index.js]: Remove blank `{" "}`

File: src/pages/LogIn/index.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                <button onClick={handleSignup}>Sign up for an account </button>{" "}
+              </p>
+            </div>
+          </form>
+        </div>
+      </div>
+      <p className={`py-4 text-center text-blue-700`}>
```
Comment (Nayeem-XTREME) [inline src/pages/LogIn/index.js]: Same

**Review by Nayeem-XTREME — CHANGES_REQUESTED** [review]

File: src/pages/LogIn/index.js (hunk 5 above)
Comment (aariful-islam) [reply src/pages/LogIn/index.js]: In form component, input field have a fixed width. I have to put same width in button. Can't fix this width with tailwind.

File: src/pages/LogIn/index.js (hunk 4 above)
Comment (aariful-islam) [reply src/pages/LogIn/index.js]: done

File: src/pages/LogIn/index.js (hunk 5 above)
Comment (aariful-islam) [reply src/pages/LogIn/index.js]: done

File: src/pages/LogIn/index.js (hunk 6 above)
Comment (aariful-islam) [reply src/pages/LogIn/index.js]: done

File: src/pages/LogIn/index.js (hunk 7 above)
Comment (aariful-islam) [reply src/pages/LogIn/index.js]: done

**Review by Nayeem-XTREME — APPROVED** [review]

**Pull request merged**